### ISSUE TITLE
Parameters are not validating?

### ISSUE THREAD

**Issue by mkgn** [oaijs/koa-oai-router-parameters#1]
Hi,

I am new to this plugin, but have managed to get it up and running using the example provided. However it doesn't seem to validate the incoming data.

Code related to koa/router is as below;

	const app = new koa();
	const router = new Router({
	  apiDoc: path.resolve(__dirname, 'api'),
	});
	
	router.mount(routerMiddleware, path.resolve(__dirname, 'controllers'));
	router.mount(new parametersHandler());

    app
	.use(bodyParser())
	.use(router.routes())

A sample swagger piece for /user/{id} route is defined as(I have remove some parts to make it short);

  '/user/{id}':
    get:
      operationId: getUserById
      x-oai-middleware:
        - file: users
          handler: getUserById
      parameters:
        - name: access_token
          in: header
          description: JWT token
          required: true
          type: string
        - name: id
          in: path
          description: 'The name that needs to be fetched. Use user1 for testing. '
          required: true
          type: integer
      responses:
        '200':
          description: OK
          schema:
            $ref: '#/definitions/ApiResponse'

And the controller method for about in /controllers directory is;

async function getUserById(ctx:koa.Context, next:Function) {
	try {
		ctx.response.status = 200;
		ctx.response.body = {}
	} catch (err) {
		console.log(err)
	}
}

Then I used developer tools to compose a request and in that I didn't provide the token and also i gave the {id} parameter a string (which should be a number). But the response is always 200 and I didn't receive any error related to invalid/missing parameter(s)

What am i missing here?

**Comment by BiteBit** [comment]
No obvious problems were found.
You can upload your demo code to the gist, so I can easily query the problem.

**Comment by mkgn** [comment]
Hi, I have shared the gist at [https://gist.github.com/mkgn/94e17c5187be7c845aa3f0e151547fda](https://gist.github.com/mkgn/94e17c5187be7c845aa3f0e151547fda)
Have remove quite a bit of code, yaml, to try and keep it small. One user.js controller is added and it has a bit of code generated by typescript. But I have removed all the other methods except for two routes.

After updating to node 10.15.2 today; when running the code it says:

(node:20744) UnhandledPromiseRejectionWarning: AssertionError [ERR_ASSERTION]: plugin must inherits from class Plugin.
    at PluginRegister.<anonymous> (E:\Projects\koatest\node_modules\koa-oai-router\lib\plugin-register.js:71:30)

So, the original error I mentioned can't be checked because of this.

I got this after the node version update. I appreciate your help!!

**Comment by BiteBit** [comment]
I found some errors in the code.

1.Mount the plugin to the router, the plugin will be executed with order of mount. If one of the plugins does not evoke next(), execution of the subsequent plugin chain will be terminated. So **koa-oai-router-parameters** must mount before **koa-oai-router-middleware**.

doc:  [router.mount(Plugin, pluginArgs)](https://github.com/BiteBit/koa-oai-router/blob/master/docs/en/references.md#routermountplugin-pluginargs)

```js
// Wrong
router.mount(routerMiddleware, path.resolve(__dirname, 'controllers'));
router.mount(parametersHandler);

// Right
router.mount(parametersHandler);
router.mount(routerMiddleware, path.resolve(__dirname, 'controllers'));
```

2.The plugin does not need to use new, it will automatically create an instance when mount.
```js
// Wrong
router.mount(new parametersHandler());

// Right
router.mount(parametersHandler);
```

3.Plugin **koa-oai-router-parameters** will crash with empty option
It's my bug, i will fix it.



This is the correct code.
```js
const Koa = require('koa');
const path = require('path');
const bodyParser = require('koa-bodyparser');
const Router = require('koa-oai-router');
const routerMiddleware = require('koa-oai-router-middleware');
const parametersHandler = require('koa-oai-router-parameters');

const app = new Koa();

const router = new Router({
  apiDoc: path.resolve(__dirname, 'api'),
});

router.mount(parametersHandler, {});
router.mount(routerMiddleware, path.resolve(__dirname, 'controllers'));

app.use(bodyParser());
app.use(router.routes());

app.listen(3000);
```

**Comment by mkgn** [comment]
Hi,

Thank you very much. I got the issue sorted.

Does this support openAPI v3.0? I was planing to use JWT Auth.. But seems
like it's only available in OpenAPI 3.0

Thanks,
Moditha

**Comment by BiteBit** [comment]
The version 3.0 is not compatible with version 2.0, and the upgrade cost is relatively large.
At present, the version 2.0 has basically met most of the usage requirements, and the 3.0 support plan has not yet been scheduled.
